week3/solutions/RO_Flowsheet_ERD_Techno_Economics.ipynb

```diff
@@ -0,0 +1,1886 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "a2a953db",
+   "metadata": {},
+   "source": [
+    "# RO Flowsheet with Energy Recovery Device Build\n",
+    "\n",
+    "This tutorial will walk you through creating a reverse osmosis (RO) flowsheet with an energy recovery device (ERD) unit model from top to bottom - building on the RO flowsheet constructed during class in RO_flowsheet_build.ipynb. \n",
+    "\n",
+    "<center><img src=\"../graphics/RO_with_ERD.png\" width=\"600\" /></center>\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5514e963",
+   "metadata": {},
+   "source": [
+    "By the end of this tutorial you should be able to follow the workflow below to construct a simple flowsheet.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1b96bd83",
+   "metadata": {},
+   "source": [
+    "## Step 1: Import Modules"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "3456f711",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Imports from Pyomo\n",
+    "from pyomo.environ import (\n",
+    "    ConcreteModel,\n",
+    "    Var,\n",
+    "    Param,\n",
+    "    Constraint,\n",
+    "    Objective,\n",
+    "    value,\n",
+    "    assert_optimal_termination,\n",
+    "    TransformationFactory,\n",
+    "    units as pyunits,\n",
+    ")\n",
+    "from pyomo.network import Arc\n",
+    "\n",
+    "# Imports from IDAES\n",
+    "from idaes.core import FlowsheetBlock\n",
+    "from idaes.models.unit_models import Feed, Product\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "from idaes.core.util.scaling import calculate_scaling_factors, set_scaling_factor\n",
+    "from idaes.core.util.initialization import propagate_state\n",
+    "\n",
+    "# Imports from WaterTAP\n",
+    "from watertap.property_models.seawater_prop_pack import SeawaterParameterBlock\n",
+    "from watertap.unit_models.pressure_changer import Pump\n",
+    "from watertap.unit_models.pressure_changer import EnergyRecoveryDevice\n",
+    "from watertap.unit_models.reverse_osmosis_0D import (\n",
+    "    ReverseOsmosis0D,\n",
+    "    ConcentrationPolarizationType,\n",
+    "    MassTransferCoefficient,\n",
+    "    PressureChangeType,\n",
+    ")\n",
+    "from watertap.core.solvers import get_solver"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0b56c5ac-9d84-4a16-b46e-a3653c441a69",
+   "metadata": {},
+   "source": [
+    "## Step 2: Build Model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "1552bd59-7d4c-46cd-8fde-84a37de98db0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a model object using ConcreteModel\n",
+    "m = ConcreteModel()"
```

Comment:
Can combine this cell with the following too. They should be getting used to this portion by now, so no need to isolate the steps